sdk/eidolon_ai_sdk/agent/api_agent.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Optional, Any, cast
+
+import jsonref
+from pydantic import BaseModel, Field
+
+from eidolon_ai_client.util.logger import logger
+from eidolon_ai_sdk.agent.agent import register_action
+from eidolon_ai_sdk.builtins.logic_units.api_helper import build_call, get_content
+from eidolon_ai_sdk.builtins.logic_units.openapi_helper import build_actions, Action, Operation
+from eidolon_ai_sdk.system.reference_model import Specable
+from eidolon_ai_sdk.util.schema_to_model import schema_to_model
+
+
+class APIAgentSpec(BaseModel):
+    title: str
+    root_call_url: str
+    open_api_location: str
+    operations_to_expose: List[Operation]  # map of tool name to openapi path
+    key_env_var: Optional[str] = None
+    key_query_param: Optional[str] = None
```

Comment:
oh, I see what these are. key query params / var are just additional query parameters and values, could we make this a generic concept were it is "additional_query_parays: dict[str, Any]" or something like that? Similar for headers would be good, then the contract simplifies a bit